lib/pbench/test/unit/server/query_apis/test_controllers_list.py

```diff
@@ -176,51 +184,39 @@ def test_query(
-        response = query_api(
-            "/controllers/list",
-            "/_search?ignore_unavailable=true",
-            json,
-            index,
-            HTTPStatus.OK,
-            status=HTTPStatus.OK,
-            json=response_payload,
-        )
-        res_json = response.json
-        assert isinstance(res_json, list)
-        assert len(res_json) == 2
-        assert res_json[0]["key"] == "unittest-controller1"
-        assert res_json[0]["controller"] == "unittest-controller1"
-        assert res_json[0]["results"] == 2
-        assert res_json[0]["last_modified_value"] == 1.59847315581e12
-        assert res_json[0]["last_modified_string"] == "2020-08-26T20:19:15.810Z"
-        assert res_json[1]["key"] == "unittest-controller2"
-        assert res_json[1]["controller"] == "unittest-controller2"
-        assert res_json[1]["results"] == 1
-        assert res_json[1]["last_modified_value"] == 1.6
-        assert res_json[1]["last_modified_string"] == "2020-09-26T20:19:15.810Z"
 
-        # Test without authorization header
-        # All tests should pass except when the non-empty user field is present in the json
-        if user != "drb":
+        # If a "user" field is present in the request json, the request
+        # need to be authenticated in order to proceed.
+        if "user" in json and json["user"] and not authentication:
+            response = client.post(
+                f"{server_config.rest_uri}/controllers/list", json=json,
+            )
+            assert response.status_code == HTTPStatus.FORBIDDEN
+
+        else:
             response = query_api(
                 "/controllers/list",
                 "/_search?ignore_unavailable=true",
                 json,
                 index,
                 HTTPStatus.OK,
-                authorization=False,
+                authorization=authentication,
                 status=HTTPStatus.OK,
                 json=response_payload,
             )
             res_json = response.json
             assert isinstance(res_json, list)
             assert len(res_json) == 2
-
-        else:
-            response = client.post(
-                f"{server_config.rest_uri}/controllers/list", json=json,
-            )
-            assert response.status_code == HTTPStatus.FORBIDDEN
+            assert res_json[0]["key"] == "unittest-controller1"
+            assert res_json[0]["controller"] == "unittest-controller1"
+            assert res_json[0]["results"] == 2
+            assert res_json[0]["last_modified_value"] == 1.59847315581e12
+            assert res_json[0]["last_modified_string"] == "2020-08-26T20:19:15.810Z"
+            assert res_json[1]["key"] == "unittest-controller2"
+            assert res_json[1]["controller"] == "unittest-controller2"
+            assert res_json[1]["results"] == 1
+            assert res_json[1]["last_modified_value"] == 1.6
+            assert res_json[1]["last_modified_string"] == "2020-09-26T20:19:15.810Z"
```

Comment:
I'm still confused as to why there is a conditional here.

For one thing, I'm concerned that the conditional is "too smart".  But, that aside, it seems to me that we should be invoking the query unconditionally (with variations controlled by `user` and `authentication`/`authorization`) and checking for the expected results.  We should be able to set the expected results (at least as far as the `response.status_code` value) using the `parameterize` decorators.  (The big block of assertions for the success case can be left substantially as is, but invoked only when the response is `OK`.)

What I'm stuck on here is that I think that the CUT should be invoked in the _same way_, modulo the individual parameterizations, and I don't feel like this code is making that happen.  In particular, we shouldn't be modifying how the query is sent based on the inputs -- it's only the expected results which should vary.  (I hope this makes sense....)